### PR TITLE
[OpenMP] Link against libm on OpenBSD

### PR DESCRIPTION
**Pull request by brad0**
Needed for some math functions in libomp.

### CODE REVIEW

File: openmp/runtime/cmake/LibompHandleFlags.cmake [hunk 1]
```diff
@@ -140,7 +140,7 @@ function(libomp_get_libflags libflags)
   if(LIBOMP_HAVE_SHM_OPEN_WITH_LRT)
     libomp_append(libflags_local -lrt)
   endif()
-  if(${CMAKE_SYSTEM_NAME} MATCHES "DragonFly|FreeBSD")
```
Comment (arsenm) [inline openmp/runtime/cmake/LibompHandleFlags.cmake:143]: OS lists like this are bad, can you use check_library_exists

Comment (shiltian) [reply openmp/runtime/cmake/LibompHandleFlags.cmake:143]: I think `check_library_exists` is not for the purpose of this case. For example, `libm` exists on a lot of systems, but we only want to link against it with proper linker flag when OS is BSD.

**Comment by brad0** [comment]
So can I commit this?

File: openmp/runtime/cmake/LibompHandleFlags.cmake (hunk 1 above)
Comment (arsenm) [reply openmp/runtime/cmake/LibompHandleFlags.cmake:143]: but why, you can do any kind of try_compile test

**Comment by brad0** [comment]
Ping.

**Review by shiltian — APPROVED** [review]
The changes look good *for now*. We definitely want to try to revise it later based on what @arsenm suggested.

**Comment by brad0** [comment]
> The changes look good _for now_. We definitely want to try to revise it later based on what @arsenm suggested.

Oh for sure. Even if you did want to use something like ```check_library_exists``` or ```check_symbol_exists``` and do something similar to how librt is handled the issue is how to test and know when to use no-as-needed / as-needed.

**Pull request merged**